src/core/execute.c

```diff
@@ -3012,6 +3012,33 @@ static int exec_parameters_get_cgroup_path(const ExecParameters *params, char **
         return using_subcgroup;
 }
 
+static int exec_context_cpu_affinity_from_numa(const ExecContext *c, CPUSet *ret) {
+        int r;
+        _cleanup_(cpu_set_reset) CPUSet s = {};
+
+        assert(c);
+        assert(ret);
+
+        if (!c->numa_policy.nodes.set) {
+                log_warning("Can't derive CPU affinity mask from NUMA mask because NUMA mask is not set, ignoring");
+                return 0;
+        }
```

Comment:
Is this warning actually needed? If the numa mask is not set, the process gets all cpus. I think this should also cause the cpuaffinity mask to be silently not set (with just a log_debug message).